mrmustard/lab/circuit_components_utils/b_to_ch.py

```diff
@@ -99,3 +102,6 @@ def inverse(self):
         ret._ansatz = super().inverse().ansatz
         ret._wires = ret.wires.dual
         return ret
+
+    def fock_array(self, shape: int | Sequence[int] | None = None) -> ComplexTensor:
+        raise NotImplementedError(f"{self.__class__.__name__} does not have a Fock representation.")
```

Comment:
Now that this is generic, you could write it in `Map`, maybe? At least, that was my thought when suggesting you write it as an f-string.

Reply:
Feel free to revert this line back (or change the other two to use `self.__class__.__name__`).